### ISSUE TITLE
Logging of auxiliary data for http/https.

### ISSUE THREAD

**Issue by johnnykv** [johnnykv/heralding#95]
Originates from #85 (@t3chn0m4g3)

**Comment by johnnykv** [comment]
Investigate which aux data we can log from the protocol. Discuss implementation and implement.

**Comment by DeveshChande** [comment]
I think logging specific headers to CSV is better as there are headers which heralding does not use, such as the ones pertaining to cookies, proxies, redirects, and DNT to name a few. Thus, I think : Date&Time, SessionID, User-Agent, Host, Username and Password will be more helpful when analyzing logs.

**Comment by johnnykv** [comment]
Sounds like a plan for starters. But i think you should ignorer username and password, not even sure they are part of the header.

**Comment by DeveshChande** [comment]
Okay, so logging the following HTTP headers is easy right off the bat : 
1. User-Agent
2. Host
3. Accept and Accept-Encoding

As far as Date and Time, the 'Date' header [is not supported in GET](https://stackoverflow.com/questions/26284553/weird-issue-with-missing-date-http-header#comment41240968_26284553) (also detailed in[ RFC 2616)](https://tools.ietf.org/html/rfc2616#section-14.18). To mitigate this, we can simply extract this data from session itself. Similarly, we can extract SessionID.

So in conclusion, we get 5 pieces of auxiliary data. Does this sound good?

**Comment by johnnykv** [comment]
In all circumstances the SessionID and timestamp should be extracted from the session, we cannot trust anything sent to us by the client. Sounds like a plan, let's see some code :-)

**Comment by johnnykv** [comment]
#102

**Comment by johnnykv** [comment]
closed with ca78d6d45cf3e6c6fcc8e98ef9aff798b52246b8
